### ISSUE TITLE
Feature Request: Add support to specify a SNS topic that belongs to a different region

### ISSUE THREAD

**Issue by DaichiUeura** [awslabs/serverless-application-model#105]
Hello,

I tried to set a SNS topic existed in a different region as a trigger like below. However CloudFormation said "Invalid parameter: TopicArn". Is this a current limitation or something like that?

```
...
Parameters:
  EventTopicArn:
    Type: String
Resources:
  SampleProcessor:
    Type: 'AWS::Serverless::Function'
    Properties:
...
      Events:
        SampleSNS:
          Type: SNS
          Properties:
            Topic: !Ref EventTopicArn
...
```

**Comment by jolexa** [comment]
@DaichiUeura - I am seeking the same answer, may you tell me why you closed this? Was there a resolution? Thanks!

**Comment by DaichiUeura** [comment]
@jolexa The reason why I closed this is that this seems to be not a SAM file issue. Here is the explanation from AWS support.

This error happens when a SNS Topic and a SNS Subscription are created in different regions(i.e. Both needs to exist in a same region). If there is a SNS Topic declaration at Events property in a SAM file, a SNS Subscription is created in the region where a stack is created.

**Comment by jolexa** [comment]
Ok, thanks. I understand and was afraid that was the case. I think that SAM should be able to parse the region from the SNS TopicArn and be able to implicitly do the correct thing. That may be tricky with the cloudformation architecture. Would you mind 1) Updating this issue's title to say "Feature Request:" and 2) re-open the issue so that we can get a SAM maintainer's opinion/attention someday?

**Comment by DaichiUeura** [comment]
@jolexa Done :)

**Comment by sanathkr** [comment]
Appreciate the feature request. SAM can't do much here because it can only meddle with resources we create in the template. Outside resources are pretty much a blackbox when it comes to SAM

**Comment by kazufumi-nishida** [comment]
I created pull request to add description in specification https://github.com/awslabs/serverless-application-model/pull/312

**Comment by evantorrie** [comment]
This should be possible now since Cloudformation has added a `Region` parameter to the SNS Topic subscription format, viz. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html

**Comment by evantorrie** [comment]
@DaichiUeura Can we reopen this?  Given that it's now supported in CloudFormation?

**Comment by keetonian** [comment]
Reopening since CFN added support.

**Comment by keetonian** [comment]
This should be a simple contribution if someone wants to make it:

Update [sns.py](https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/sns.py) to match the CFN equivalent in the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html)

Add `Region` into the [SNS Event](https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/eventsources/push.py#L341) definition.

Add appropriate tests and documentation.

**Comment by 53ningen** [comment]
I created a pull request to resolve this issue.
https://github.com/awslabs/serverless-application-model/issues/105#issuecomment-476771718

**Comment by jlhood** [comment]
Pending v1.14.0 release

**Comment by ShreyaGangishetty** [comment]
Closing this issue as this is released in v1.14.0
